assets/app/scripts/controllers/create.js

```diff
@@ -94,6 +107,213 @@ angular.module('openshiftConsole')
+        createAndUpdate();
+      });
+    }
+
+    // create 
+    function createAndUpdate() {
+      $scope.createdAndUpdateResources = 0;
+
+      if (!_.isEmpty($scope.updateResources)) {
+        angular.forEach($scope.updateResources, function(item) {
+          DataService.update(APIService.kindToResource(item.kind), item.metadata.name, item, {namespace: $scope.projectName}).then(
+            // update resource success
+            function() {
+              AlertMessageService.addAlert({
+                name: item.metadata.name,
+                data: {
+                  type: "success",
+                  message: item.kind + " " + item.metadata.name + " was successfully updated."
+                }
+              });
+              redirectIfComplete();
+            },
+            // update resource failure
+            function(result) {
+              AlertMessageService.addAlert({
+                name: item.metadata.name,
+                data: {
+                  type: "error",
+                  message: "An error occurred updating the " + item.kind + " " + item.metadata.name,
+                  details: $filter('getErrorDetails')(result)
+                }
+              });
+              redirectIfComplete();
+            });
+        });
+      }
+
+      if (!_.isEmpty($scope.createResources)) {
+        angular.forEach($scope.createResources, function(item) {
+          DataService.create(APIService.kindToResource(item.kind), null, item, {namespace: $scope.projectName}).then(
+            // create resource success
+            function() {
+              AlertMessageService.addAlert({
+                name: item.metadata.name,
+                data: {
+                  type: "success",
+                  message: item.kind + " " + item.metadata.name + " was successfully created."
+                }
+              });
+              redirectIfComplete();
+            },
+            // create resource failure
+            function(result) {
+              AlertMessageService.addAlert({
+                name: item.metadata.name,
+                data: {
+                  type: "error",
+                  message: "An error occurred creating the " + item.kind + " " + item.metadata.name,
+                  details: $filter('getErrorDetails')(result)
+                }
+              });
+              redirectIfComplete();
+            });
+        });
+      }  
+    }
+
+    // Redirect to appropriate page based on the created/updated resource. If resource is Template redirect to
+    // frotemplate page, else to the project overview.
+    function redirectIfComplete() {
+      $scope.createdAndUpdateResources++;
+      if ($scope.createdAndUpdateResources === $scope.resourceList.length) {
+        var subPath;
+        if ($scope.resourceKind === "Template") {
+          subPath =  "create/fromtemplate?name=" + $scope.resourceName + "&namespace=" + encodeURIComponent($scope.projectName);
```

Comment:
`$scope.resourceKind` is holding the top level resource Kind. So in the case of TemplateList it will be redirected to overview